src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -262,7 +274,13 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
-        Collections.singleton(Modifier.STATIC));
+        Collections.singleton(Modifier.STATIC)),
+
+    ANNOTATION(
+        Collections.<Modifier>emptySet(),
```

Comment:
This should be the same as for regular interfaces. Annotation fields work just like interface fields with respect to default modifiers.